include/rocksdb/uuid.h

```diff
@@ -0,0 +1,117 @@
+//  Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <cstdint>
+
+#include "rocksdb/status.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+// A base struct / holder for an identifier of up to 128 bits. All zeros is
+// considered empty, so implementations should avoid generating / saving the
+// empty value when a value is actually generated.
+struct RawUuid {
+  //== Data
+  uint64_t lower64 = 0;
+  uint64_t upper64 = 0;
+
+  //== Construction
+  RawUuid() { assert(IsEmpty()); }
+
+  static RawUuid FromRawBytes(const char *buf_of_16);
+
+  //== Functions
+  inline bool IsEmpty() const { return lower64 == 0 && upper64 == 0; }
+
+  // For saving into a machine-independent 16-byte string. The written
+  // characters can contain nul bytes.
+  std::string ToRawBytes() const;
+  // For saving into a machine-independent 16-byte string, into an
+  // existing buffer. No trailing nul is written or expected, but the
+  // written characters can contain nul bytes. Returns a Slice for the
+  // buffer and 16 written characters.
+  Slice PutRawBytes(char *buf_of_16) const;
```

Comment:
Please better explain the arguments and return value.  From the signature, I presume the Slice is mapped over buf_of_16, but it would be nice to explicitly state that.